subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java

```diff
@@ -192,7 +192,7 @@ WatchingAwareVirtualFileSystem createVirtualFileSystem(
                 .orElse(new NonWatchingVirtualFileSystem(delegate));
             listenerManager.addListener(new VirtualFileSystemBuildLifecycleListener(
                 watchingAwareVirtualFileSystem,
-                startParameter -> isRetentionEnabled(startParameter.getSystemPropertiesArgs()),
+                StartParameter::isWatchFileSystem,
```

Comment:
Can we remove the method now?

Reply:
Nearly. We still use the method for checking whether the property is set, so we can emit a deprecation warning.